plugins/modules/java_cert.py

```diff
@@ -569,16 +579,22 @@ def main():
             if alias_exists:
                 # The certificate in the keystore does not match with the one we want to be present
                 # The existing certificate must first be deleted before we insert the correct one
-                delete_cert(module, executable, keystore_path, keystore_pass, cert_alias, keystore_type, exit_after=False)
+                delete_cert(module, executable, keystore_path, keystore_pass, cert_alias, keystore_type)
 
             if pkcs12_path:
-                import_pkcs12_path(module, executable, pkcs12_path, pkcs12_pass, pkcs12_alias,
-                                   keystore_path, keystore_pass, cert_alias, keystore_type)
+                result = import_pkcs12_path(module, executable, pkcs12_path, pkcs12_pass, pkcs12_alias,
+                                            keystore_path, keystore_pass, cert_alias, keystore_type)
             else:
-                import_cert_path(module, executable, new_certificate, keystore_path,
-                                 keystore_pass, cert_alias, keystore_type, trust_cacert)
+                result = import_cert_path(module, executable, new_certificate, keystore_path,
+                                          keystore_pass, cert_alias, keystore_type, trust_cacert)
+
+    if os.path.exists(keystore_path):
+        if 'changed' in result:
+            result['changed'] = result['changed'] or _update_permissions(module, keystore_path)
+        else:
+            result['changed'] = _update_permissions(module, keystore_path)
```

Comment:
In line 593, `_update_permissions` will only be called if `result['changed']` is `False`. Also there's no need for the `if`. Better write this as follows:
```suggestion
        changed_permissions =_update_permissions(module, keystore_path)
        result['changed'] = result.get('changed', False) or changed_permissions
```

Reply:
Yes, it looks more elegant. Thank you